src/app/search/search-filters/sockets.tsx

```diff
@@ -222,7 +222,11 @@ const socketFilters: FilterDefinition[] = [
           case 'incomplete':
             return !item.deepsightInfo.attunementObjective.complete;
           case 'pattern':
-            return item.deepsightInfo.extractPattern;
+            return Boolean(
+              !item.deepsightInfo.attunementObjective.complete &&
+                item.patternUnlockRecord &&
+                item.patternUnlockRecord.state & DestinyRecordState.ObjectiveNotCompleted
+            );
```

Comment:
Similarly to the tooltip, why match only if the objective is incomplete? I'd love to know if I have a gun with a completed objective in my vault I can use to make pattern progress.

Reply:
Fair enough - do people leave completed objective items lying around? I guess because of the materials cap?